src/c++/perf_analyzer/genai-perf/genai_perf/graphs/scatter_plot.py

```diff
@@ -0,0 +1,98 @@
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import pandas as pd
+import plotly.express as px
+from genai_perf.graphs.base_plot import BasePlot
+from genai_perf.llm_metrics import Statistics
+
+
+class ScatterPlot(BasePlot):
+    """
+    Generate a scatter plot in jpeg and html format.
+    """
+
+    def __init__(self, stats: Statistics) -> None:
+        super().__init__(stats)
+
+    def create_scatter_plot(
+        self,
+        x_key: str = "",
+        scale_x: bool = False,
+        y_key: str = "",
+        scale_y: bool = False,
+        graph_title: str = "",
+        x_label: str = "",
+        y_label: str = "",
+        filename_root: str = "",
+        preprocessed_x_data=[],
+        preprocessed_y_data=[],
+    ):
+        if not preprocessed_x_data and not preprocessed_y_data:
+            if scale_x:
+                x_values = [
+                    self._scale(x, (1 / 1e9)) for x in self._metrics.data[x_key]
+                ]
+            else:
+                x_values = self._metrics.data[x_key]
+            if scale_y:
+                y_values = [
+                    self._scale(x, (1 / 1e9)) for x in self._metrics.data[y_key]
+                ]
+            else:
+                y_values = self._metrics.data[y_key]
+        else:
+            x_values = preprocessed_x_data
+            y_values = preprocessed_y_data
+
+        df = pd.DataFrame(
+            {
+                x_key: x_values,
+                y_key: y_values,
+            }
+        )
+
+        fig = px.scatter(
+            df,
+            x=x_key,
+            y=y_key,
+            trendline="ols",
```

Comment:
Is there a way to label the trendline to indicate that's what it is? Also, if it would be easy, it could be nice to provide the equation for this. If not easy, we should probably not do it until consulting with an SA in a follow-up feature, if there is enough demand. For example, it'd be good to know that the time-to-first-token is approximately equal to the number input tokens + .002 seconds per token. That would let customers calculate the expected time-to-first-token for their deeper analyses.

Reply:
I think we need a ticket to investigate this.

Reply:
Understood. Let's resolve it as this is more of a "nice-to-have" and probably not worth it if the library doesn't offer it.